modules/carto/src/api/fetch-map.ts

```diff
@@ -16,7 +16,7 @@ import {
   vectorQuerySource,
   vectorTableSource,
   vectorTilesetSource
-} from '../sources';
+} from '../sources/index';
```

Comment:
We could avoid all these `'xxxxxx/index'` imports by using [experimentalSpecifierResolution](https://typestrong.org/ts-node/api/interfaces/TsConfigOptions.html#experimentalSpecifierResolution) in tsconfig

Reply:
Note that these import paths don't just affect source code, they also persist into `*.d.ts` files, and will break depending on a downstream application's TSConfig. See:

- https://github.com/visgl/luma.gl/issues/1861
- https://github.com/developit/microbundle/issues/1019#issuecomment-1384721680
- https://github.com/donmccurdy/glTF-Transform/issues/824

Ideally — I think we should either go one step further and add the extensions, like `../sources/index.js`, or else take @ibgreen's suggestion from https://github.com/visgl/luma.gl/issues/1861#issuecomment-1839471289 and post-process the `*.d.ts` files.

Reply:
.js extension is added with a custom typescript transform, hence `xxx/index` which is processed into `xxx/index.js`.

We could alternatively write all the extensions into src, I was trying to avoid too many changes.

Reply:
I'd missed the custom TS transform — that sounds like a good solution to me!

One other option would be to bundle the type definitions with something like tsup or dts-buddy — no need for extensions if there are no imports! — but if the custom TS transform is working already, I'd vote for that. 👍🏻